benchmark/test/test_operations.py

```diff
@@ -0,0 +1,23 @@
+from zope.interface.verify import verifyObject
+
+from twisted.trial.unittest import TestCase
+
+from benchmark.benchmark_operations import _operations, IProbe, IOperation
+
+
+def check_interfaces(factory):
+
+    class OperationTests(TestCase):
+
+        def test_interfaces(self):
+            operation = factory(client=None)
+            verifyObject(IOperation, operation)
+            probe = operation.get_probe()
+            verifyObject(IProbe, probe)
+
+    testname = '{}InterfaceTests'.format(factory.__name__)
+    OperationTests.__name__ = testname
+    globals()[testname] = OperationTests
+
+for factory in _operations.values():
+    check_interfaces(factory)
```

Comment:
I simplified it a bit, but this is how zope interfaces are meant to be tested, and the cleverness attempts to create a test for each registered operation, without burdening future developers.